src/components/ContentOpenWith/ContentOpenWith.js

```diff
@@ -298,6 +364,30 @@ class ContentOpenWith extends PureComponent<Props, State> {
         this.integrationWindow = null;
     };
 
+    /**
+     * Opens the file via Box Edit
+     *
+     * @private
+     * @param {ExecuteAPI} executeData - API response on how to open an executed integration
+
+     * @return {void}
+     */
+    executeBoxEditSuccessHandler = (executeData: ExecuteAPI): void => {
+        const { fileId, token, onExecute } = this.props;
+        const { url } = executeData;
+        const authCode = url.split(AUTH_CODE_DELIMITER)[1];
+
+        this.boxEdit.openFile(`f_${fileId}`, {
```

Comment:
Since box edit is considered an API, shouldn't this be inside the api layer?

Reply:
do you mean adding it to the factory? I don't want to refactor the library itself yet since we don't own it.